### ISSUE TITLE
Support arm64 machine

### ISSUE THREAD

**Issue by isac322** [paambaati/codeclimate-action#663]
**Describe the bug**

The actions do not work on arm64 machine.

```
/runner/_work/redirector/redirector/cc-reporter: 1: ELF: not found
/runner/_work/redirector/redirector/cc-reporter: 1: ��: not found
/runner/_work/redirector/redirector/cc-reporter: 1: ��: not found
/runner/_work/redirector/redirector/cc-reporter: 1: �����c���c��c��$: not found
/runner/_work/redirector/redirector/cc-reporter: 1: cannot open : No such file
/runner/_work/redirector/redirector/cc-reporter: 1: ���P���Q�k�: not found
/runner/_work/redirector/redirector/cc-reporter: 2: Syntax error: word unexpected (expecting ")")
```

**Version of `codeclimate-action` you're using**
`v3.2.0`

**Comment by paambaati** [comment]
@isac322 Can you show me how this is run? AFAICT, Action runners do not run on ARM (see https://github.com/actions/runner-images/issues/5631).

**Comment by paambaati** [comment]
@isac322 Looking at the linked repo, I’m assuming your runners are self-hosted. Can you show me how Node.js is installed?

I ask because detecting the host platform architecture is not straightforward, as Node `os.arch()` only exposes the architecture Node itself was compiled for, and not of the underlying host.

**Comment by isac322** [comment]
@paambaati 

```yaml
- name: debug
  run: |
    /runnertmp/externalstmp/node12/bin/node --version
    /runnertmp/externalstmp/node12/bin/node -e 'console.log(require("os").arch())'
    /runnertmp/externalstmp/node16/bin/node --version
    /runnertmp/externalstmp/node16/bin/node -e 'console.log(require("os").arch())'
```

![image](https://user-images.githubusercontent.com/4602209/233372285-7b688af8-543c-4e91-b092-b6255b9f5275.png)

You can reproduce using `ghcr.io/actions-runner-controller/actions-runner-controller/actions-runner` on local. Please post any code if you need to run on arm64.

**Comment by paambaati** [comment]
This should now be available in v4.0.0 of the workflow.
